aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/service/AWSS3StorageService.java

```diff
@@ -147,6 +150,27 @@ public TransferObserver uploadFile(
         return transferUtility.upload(bucket, serviceKey, file, metadata);
     }
 
+    /**
+     * Begin uploading an inputStream.
+     * @param serviceKey S3 service key
+     * @param inputStream Target inputStream
+     * @param metadata Object metadata to associate with upload
+     * @return A transfer observer
+     * @throws IOException An IOException thrown during the process writing an inputStream into a file
+     */
+    @NonNull
+    public TransferObserver uploadInputStream(
+            @NonNull String serviceKey,
+            @NonNull InputStream inputStream,
+            @NonNull ObjectMetadata metadata
+    ) throws IOException {
+        startServiceIfNotAlreadyStarted();
+        return transferUtility.upload(serviceKey, inputStream, UploadOptions.builder()
+                                                                            .bucket(bucket)
+                                                                            .objectMetadata(metadata)
+                                                                            .build());
```

Comment:
This formatting would probably be ceaner:

```suggestion
        UploadOptions uploadOptions = UploadOptions.builder()
            .bucket(bucket)
            .objectMetadata(metadata)
            .build()
        return transferUtility.upload(serviceKey, inputStream, uploadOptions);
```